network/modules/UDP/server_socket.py

```diff
@@ -0,0 +1,59 @@
+import socket
+from network.modules.UDP.socket_wrapper import UdpSocket
+
+
+class UdpServerSocket(UdpSocket):
+    """
+    Wrapper for server socket operations.
+    """
+
+    __create_key = object()
+
+    def __init__(
+        self,
+        class_private_create_key: object,
+        socket_instance: socket.socket,
+        server_address: tuple,
+    ) -> None:
+        """
+        Private Constructor, use create() method.
+        """
+        assert class_private_create_key is UdpServerSocket.__create_key, "Use create() method"
+        super().__init__(socket_instance=socket_instance)
+        self.__socket = socket_instance
+        self.server_address = server_address
+
+    @classmethod
+    def create(cls, host: str = "", port: int = 5000) -> "tuple[bool, UdpServerSocket | None]":
```

Comment:
Can we add a parameter for setting a timeout? Have a default of like 10seconds (see `settimeout()`)